### PR TITLE
Deprecate Google+ APIs

### PR DESCRIPTION
**Pull request by gonzaloriestra**
Related: https://github.com/CartoDB/product/issues/196

There's a new endpoint to retrieve the user data and it comes in a different format, so I had to change the parser. More info about the deprecation: https://developers.google.com/+/api-shutdown

I've also made some cleaning:
- Removed GooglePlus helpers, as it's enough with `Oauth::Google::Api`
- Removed `/google_plus` endpoint (it wasn't working)
- Removed revoke functionality (it wasn't working)
- Renamed some code from `google_plus` to just `google` (except the config parameter from `app_config.yml` to avoid breaking compatibility)

Similar changes also made in https://github.com/CartoDB/cartodb-central/pull/2409

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -373,7 +365,7 @@ def user_password_authentication?
   end
 
   def google_authentication?
-    params[:google_access_token].present? && @google_plus_config.present?
```
Comment (alrocar) [inline app/controllers/sessions_controller.rb]: not sure about this, what happens if there's no google config?

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -350,16 +348,10 @@ def saml_strategy_username
 
   def google_strategy_username
     if google_authentication? && !user_password_authentication?
-      user = GooglePlusAPI.new.get_user(params[:google_access_token])
+      api = Carto::Oauth::Api::Google.new(nil, params[:google_access_token])
```
[the remark this answers is not in the document]
Comment (gonzaloriestra) [reply app/controllers/sessions_controller.rb]: Yes, it is :man_facepalming:. But this is unused, as the Oauth flow goes through `OauthLoginController` instead :fire:

File: lib/user_account_creator.rb [hunk 3]
```diff
@@ -127,8 +126,8 @@ def user
 
     def with_google_token(google_access_token)
       @built = false
-      # get_user_data can return nil
-      @google_user_data = GooglePlusAPI.new.get_user_data(google_access_token)
+      api = Carto::Oauth::Api::Google.new(nil, google_access_token)
```
[the remark this answers is not in the document]
Comment (gonzaloriestra) [reply lib/user_account_creator.rb]: Also unused :fire:

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (gonzaloriestra) [reply app/controllers/sessions_controller.rb]: This is removed now, but when there's no google config, the google button is not even shown.

**Comment by gonzaloriestra** [comment]
Regarding the specs, I agree it was worrying that nothing broke with the wrong modules, but it was unused code. I've added some specs for the new parsing, and the callback seems to be well covered, so I think it's enough.

**Comment by alrocar** [comment]
Acceptance 🍏 

- Org signups
- Org admin > Create user (with gmail e-mail)
- Org admin > Invite user (with gmail e-mail)

**Pull request merged**